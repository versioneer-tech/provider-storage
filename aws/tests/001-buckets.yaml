---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: alice-001
  namespace: storage
  uid: xyz
spec:
  buckets:
  - bucketName: alice-001
    discoverable: false
  - bucketName: alice-shared-001
    discoverable: true
  owner: alice-001
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-alice-001, bucket-alice-001, bucket-alice-shared-001,
      and 5 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider:
    userRef:
      name: alice-001
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
  writeConnectionSecretToRef:
    name: alice-001
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-shared-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-shared-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-owner-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001.owner.alice-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::alice-001",
              "arn:aws:s3:::alice-001/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-owner-alice-shared-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001.owner.alice-shared-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::alice-shared-001",
              "arn:aws:s3:::alice-shared-001/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-owner-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001.owner.alice-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider:
    policyArnRef:
      name: alice-001.owner.alice-001
    userRef:
      name: alice-001
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-owner-alice-shared-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001.owner.alice-shared-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider:
    policyArnRef:
      name: alice-001.owner.alice-shared-001
    userRef:
      name: alice-001
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-alice-001
  labels:
    crossplane.io/composite: alice-001
  name: alice-001
  namespace: storage
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-001
    uid: xyz
spec:
  forProvider: {}
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
