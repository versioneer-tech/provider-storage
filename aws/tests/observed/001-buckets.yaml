apiVersion: kubernetes.crossplane.io/v1alpha1
kind: Object
metadata:
  name: observe-joe-restricted-s-jeff-shared
  annotations:
    crossplane.io/composition-resource-name: observe-joe-restricted-s-jeff-shared
status:
  atProvider:
    manifest:
      apiVersion: iam.aws.m.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: joe.restricted.s-jeff-shared
  conditions:
  - type: Ready
    status: "True"