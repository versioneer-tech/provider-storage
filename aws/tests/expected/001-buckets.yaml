---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-joe
spec:
  bucketAccessGrants:
  - bucketName: s-joe
    grantedAt: "2025-09-29T10:05:00Z"
    grantee: jeff
    permission: ReadWrite
  bucketAccessRequests:
  - bucketName: s-jeff-shared
    reason: Need access
    requestedAt: "2025-09-29T10:00:00Z"
  buckets:
  - bucketName: s-joe
    discoverable: true
  principal: joe
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-joe, bucket-s-joe, observe-joe-restricted-s-jeff-shared,
      and 4 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-joe
  labels:
    crossplane.io/composite: s-joe
  name: joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    userRef:
      name: joe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
  writeConnectionSecretToRef:
    name: joe
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-joe-restricted-s-jeff-shared
  labels:
    crossplane.io/composite: s-joe
  name: observe-joe-restricted-s-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.m.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: joe.restricted.s-jeff-shared
        namespace: default
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-full-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: joe.full.s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::s-joe",
              "arn:aws:s3:::s-joe/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-jeff-restricted-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: jeff.restricted.s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:ListBucket",
              "s3:GetObject",
              "s3:PutObject",
              "s3:DeleteObject"
            ],
            "Resource": [
              "arn:aws:s3:::s-joe",
              "arn:aws:s3:::s-joe/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-full-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: joe.full.s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: joe.full.s-joe
    userRef:
      name: joe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-joe
  labels:
    crossplane.io/composite: s-joe
  name: joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider: {}
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
